File: back/src/main/java/com/back/global/security/filter/BearerTokenExtractor.java
package com.back.global.security.filter;

import com.back.global.exception.ErrorCode;
import com.back.global.exception.ServiceException;
import org.springframework.stereotype.Component;

@Component
public class BearerTokenExtractor {
    private static final String BEARER_PREFIX = "Bearer ";

    public String extract(String authorization) {
        if (!authorization.startsWith(BEARER_PREFIX)) {
            throw new ServiceException(ErrorCode.AUTH_INVALID_BEARER_HEADER);
        }

        String accessToken = authorization.substring(BEARER_PREFIX.length()).trim();

        if (accessToken.isBlank()) {
            throw new ServiceException(ErrorCode.AUTH_INVALID_BEARER_HEADER);
        }

        return accessToken;
    }
}

File: back/src/main/kotlin/com/back/global/security/filter/BearerTokenExtractor.kt
package com.back.global.security.filter

import com.back.global.exception.ErrorCode
import com.back.global.exception.ServiceException
import org.springframework.stereotype.Component

@Component
class BearerTokenExtractor {

    fun extract(authorization: String): String =
        authorization.takeIf { it.startsWith(BEARER_PREFIX) }
            ?.removePrefix(BEARER_PREFIX)
            ?.trim()
            ?.takeIf { it.isNotBlank() }
            ?: throw ServiceException(ErrorCode.AUTH_INVALID_BEARER_HEADER)

    fun extractAccessTokenOrNull(authorization: String?): String? =
        authorization?.takeIf { it.startsWith(BEARER_PREFIX) }
            ?.removePrefix(BEARER_PREFIX)
            ?.trim()
            ?.takeIf { it.isNotBlank() }

    companion object {
        private const val BEARER_PREFIX = "Bearer "
    }
}
